panic on InitWallet restore:true with passphrase different than TDEX_WALLET_UNLOCK_PASSWORD_FILE set

```
time="2022-08-02T21:32:17Z" level=info msg="All 0 tables opened in 0s\n"
time="2022-08-02T21:32:17Z" level=info msg="Replaying file id: 0 at offset: 0\n"
time="2022-08-02T21:32:17Z" level=info msg="Replay took: 41.201µs\n"
time="2022-08-02T21:32:17Z" level=info msg="All 0 tables opened in 0s\n"
time="2022-08-02T21:32:17Z" level=info msg="Replaying file id: 0 at offset: 0\n"
time="2022-08-02T21:32:17Z" level=info msg="Replay took: 4.801µs\n"
time="2022-08-02T21:32:17Z" level=info msg="All 0 tables opened in 0s\n"
time="2022-08-02T21:32:17Z" level=info msg="Replaying file id: 0 at offset: 0\n"
time="2022-08-02T21:32:17Z" level=info msg="Replay took: 4.803µs\n"
time="2022-08-02T21:32:17Z" level=info msg="rescan internal wallet's utxo set. This could take a while..."
time="2022-08-02T21:32:17Z" level=debug msg="Restored uspents: 0"
time="2022-08-02T21:32:17Z" level=debug msg="Restore took: 0.00s"
time="2022-08-02T21:32:17Z" level=info msg=done
time="2022-08-02T21:32:17Z" level=info msg="starting daemon"
time="2022-08-02T21:32:17Z" level=info msg="wallet unlocker interface is listening on :9000"
time="2022-08-02T21:32:17Z" level=debug msg="stop grpc-web Operator server"
time="2022-08-02T21:32:17Z" level=debug msg="stop grpc Operator server"
time="2022-08-02T21:32:17Z" level=debug msg="stop mux Operator"
panic: passphrase provided is not correct

goroutine 121 [running]:
github.com/tdex-network/tdex-daemon/internal/interfaces/grpc.(*service).startListeningToReadyChan(0xc010334000)
	/home/runner/work/tdex-daemon/tdex-daemon/internal/interfaces/grpc/service.go:482 +0x3d0
created by github.com/tdex-network/tdex-daemon/internal/interfaces/grpc.(*service).Start
	/home/runner/work/tdex-daemon/tdex-daemon/internal/interfaces/grpc/service.go:277 +0x10a
time="2022-08-02T21:32:47Z" level=info msg="All 0 tables opened in 0s\n"
time="2022-08-02T21:32:47Z" level=info msg="Replaying file id: 0 at offset: 0\n"
time="2022-08-02T21:32:47Z" level=info msg="Replay took: 34.603µs\n"
time="2022-08-02T21:32:47Z" level=info msg="All 0 tables opened in 0s\n"
time="2022-08-02T21:32:47Z" level=info msg="Replaying file id: 0 at offset: 0\n"
time="2022-08-02T21:32:47Z" level=info msg="Replay took: 4.78µs\n"
time="2022-08-02T21:32:47Z" level=info msg="All 0 tables opened in 0s\n"
time="2022-08-02T21:32:47Z" level=info msg="Replaying file id: 0 at offset: 0\n"
time="2022-08-02T21:32:47Z" level=info msg="Replay took: 9.42µs\n"
time="2022-08-02T21:32:47Z" level=info msg="rescan internal wallet's utxo set. This could take a while..."
time="2022-08-02T21:32:47Z" level=debug msg="Restored uspents: 0"
time="2022-08-02T21:32:47Z" level=debug msg="Restore took: 0.00s"
time="2022-08-02T21:32:47Z" level=info msg=done
time="2022-08-02T21:32:47Z" level=info msg="starting daemon"
time="2022-08-02T21:32:47Z" level=info msg="wallet unlocker interface is listening on :9000"
time="2022-08-02T21:32:47Z" level=debug msg="stop grpc-web Operator server"
time="2022-08-02T21:32:47Z" level=debug msg="stop grpc Operator server"
time="2022-08-02T21:32:47Z" level=debug msg="stop mux Operator"
panic: passphrase provided is not correct

goroutine 97 [running]:
github.com/tdex-network/tdex-daemon/internal/interfaces/grpc.(*service).startListeningToReadyChan(0xc00062c480)
	/home/runner/work/tdex-daemon/tdex-daemon/internal/interfaces/grpc/service.go:482 +0x3d0
created by github.com/tdex-network/tdex-daemon/internal/interfaces/grpc.(*service).Start
	/home/runner/work/tdex-daemon/tdex-daemon/internal/interfaces/grpc/service.go:277 +0x10a
```

--- a/internal/interfaces/grpc/service.go
+++ b/internal/interfaces/grpc/service.go
@@ -523,7 +523,6 @@ func (s *service) startListeningToReadyChan() {
 
 	if s.walletPassword != "" {
 		retryAttempt := 0
-	loop:
 		for {
 			if err := s.opts.WalletUnlockerSvc.UnlockWallet(
 				context.Background(),
@@ -532,7 +531,7 @@ func (s *service) startListeningToReadyChan() {
 				retryAttempt++
 				log.Debugf("retrying to unlock wallet: %v, attempt: %v", err, retryAttempt)
 
-				continue loop
+				continue
 			} else {
 				if err != nil {
 					log.WithError(err).Warn(
@@ -542,7 +541,7 @@ func (s *service) startListeningToReadyChan() {
 					log.Debug("wallet unlocked with password file")
 				}
 
-				break loop
+				break
 			}
 		}
 	}

--- a/internal/interfaces/grpc/service.go
+++ b/internal/interfaces/grpc/service.go
@@ -11,6 +11,8 @@ import (
 	"net/http"
 	"path/filepath"
 
+	"github.com/tdex-network/tdex-daemon/pkg/wallet"
+
 	"github.com/tdex-network/tdex-daemon/internal/core/ports"
 
 	httpinterface "github.com/tdex-network/tdex-daemon/internal/interfaces/http"
@@ -520,12 +522,29 @@ func (s *service) startListeningToReadyChan() {
 	}
 
 	if s.walletPassword != "" {
-		if err := s.opts.WalletUnlockerSvc.UnlockWallet(context.Background(), s.walletPassword); err != nil {
-			panic(err)
-		}
+		retryAttempt := 0
+	loop:
+		for {
+			if err := s.opts.WalletUnlockerSvc.UnlockWallet(
+				context.Background(),
+				s.walletPassword,
+			); err != nil && err != wallet.ErrInvalidPassphrase && retryAttempt < 5 {
+				retryAttempt++
+				log.Debugf("retrying to unlock wallet: %v, attempt: %v", err, retryAttempt)
 
-		s.walletPassword = ""
-		log.Infoln("wallet unlocked")
+				continue loop
+			} else {
+				if err != nil {
+					log.WithError(err).Warn(
+						"an error occurred while unlocking wallet",
+					)
+				} else {
+					log.Debug("wallet unlocked with password file")
+				}
+
+				break loop
+			}
+		}
 	}
 
 	withoutUnlockerOnly := false

--- a/internal/interfaces/grpc/service.go
+++ b/internal/interfaces/grpc/service.go
@@ -544,6 +544,7 @@ func (s *service) startListeningToReadyChan() {
 				break
 			}
 		}
+		s.walletPassword = ""
 	}
 
 	withoutUnlockerOnly := false

